lib/TPP/ConvertTppToLoops.cpp

```diff
@@ -71,6 +71,30 @@ struct ConvertTppAddOp : public OpRewritePattern<AddOp> {
   }
 };
 
+void buildUnaryLoop(
+    PatternRewriter &rewriter, TppOp tppOp,
+    function_ref<void(OpBuilder &, Location, ValueRange)> bodyBuilder) {
+  assert(tppOp.isUnary() && "Expect tpp unary op");
+
+  Location loc = tppOp.getLoc();
+
+  SmallVector<Value> ubs;
+  size_t rank = tppOp.getOutput().getType().cast<MemRefType>().getRank();
+  ArrayRef<int64_t> shapeOutput =
+      tppOp.getOutput().getType().cast<MemRefType>().getShape();
+  for (size_t idx = 0; idx < rank; idx++) {
+    Value dim = rewriter.create<arith::ConstantIndexOp>(loc, shapeOutput[idx]);
```

Comment:
I would avoid creating the temp

Reply:
I'm not sure what would be the alternative?

Reply:
IIUC, this isn't a temp, it's just indices to the loop.